dev/wlp.lib.extract/src/wlp/lib/extract/SelfExtract.java

```diff
@@ -167,6 +175,15 @@ public static int doMain(String[] args) {
             } else {
                 outputDirFromUser = new File(targetString.trim());
             }
+
+            if (outputDirFromUser.getPath().indexOf('~') >= 0) {
+                String pathName = outputDirFromUser.getPath();
+                String pathNameToReturn = pathName.replaceAll("~", "");
```

Comment:
Should check for `~` only at the beginning of the string, since `~` is a legal character in the middle of a path (e.g. it could be part of a directory name).  Then in that case, don't do `replaceAll` but only replace the first instance of that char.